### DISCUSSION TITLE
Can erlang version be updated on live cluster?

### DISCUSSION THREAD

**Discussion by bdoyle0182** [apache/couchdb#4362] [Q&A] [answered]
Just assume I'm a newb with limited knowledge of how erlang works under the hood. Simply can I go from erlang 20.3.8.x to 23.x through rolling upgrade of the cluster? Or could I incrementally go through erlang major versions until I get to 23.x?

Or is the only option to create a new cluster with the upgraded erlang and replicate old cluster -> new cluster

**Comment by nickva** [comment, marked as answer]
It should be possible. We upgraded from 20 to 23 at Cloudant at one point using rolling upgrades. Usually the OTP team promises to be compatible up to 2 releases back. So 23 would be compatible with 22 and 21. But they also don't go out of the way to break compatibility and 20 -> 23 seemed to have worked.
